DirectProgramming/DPC++FPGA/Tutorials/Features/scheduler_target_fmax/README.md

```diff
@@ -0,0 +1,147 @@
+# Scheduler Target FMAX
+This tutorial explains the `scheduler_target_fmax_mhz` attribute and its effect on the performance of Intel® FPGA kernels.
+
+***Documentation***:  The [DPC++ FPGA Code Samples Guide](https://software.intel.com/content/www/us/en/develop/articles/explore-dpcpp-through-intel-fpga-code-samples.html) helps you to navigate the samples and build your knowledge of DPC++ for FPGA. <br>
+The [oneAPI DPC++ FPGA Optimization Guide](https://software.intel.com/content/www/us/en/develop/documentation/oneapi-fpga-optimization-guide) is the reference manual for targeting FPGAs through DPC++. <br>
+The [oneAPI Programming Guide](https://software.intel.com/en-us/oneapi-programming-guide) is a general resource for target-independent DPC++ programming.
+
+| Optimized for                     | Description
+---                                 |---
+| OS                                | Linux* Ubuntu* 18.04/20.04, RHEL*/CentOS* 8, SUSE* 15; Windows* 10
+| Hardware                          | Intel® Programmable Acceleration Card (PAC) with Intel Arria® 10 GX FPGA <br> Intel® FPGA Programmable Acceleration Card (PAC) D5005 (with Intel Stratix® 10 SX) <br> Intel® FPGA 3rd party / custom platforms with oneAPI support <br> *__Note__: Intel® FPGA PAC hardware is only compatible with Ubuntu 18.04*
+| Software                          | Intel® oneAPI DPC++ Compiler <br> Intel® FPGA Add-On for oneAPI Base Toolkit
+| What you will learn               |  The behavior of the `scheduler_target_fmax_mhz` attribute and when to use it on your kernel <br> The effect this attribute can have on your kernel's performance on FPGA
+| Time to complete                  | 15 minutes
+
+## Purpose
+This tutorial demonstrates how to use `[[intel::scheduler_target_fmax_mhz(N)]]` attribute to direct the compiler to prioritize a high fMAX over a low II.
+
+A bottleneck in a loop is one or more loop-carried dependencies that cause the loop to have either an II greater than one, or a lower fMAX to achieve II of one. By default, the compiler finds the fMAX-II combination that provides the best fMAX-to-II ratio.
+
+The compiler provides `[[intel::scheduler_target_fmax_mhz(N)]]` attribute to set your desired fMAX. When you specify the fMAX, the compiler schedules the design at that specified fMAX value with minimal achievable II.
+
+In the tutorial, two kernels implement the same function (BKDR Hash).
+In kernel `NoSchedulerTargetFMAX`, no fMAX constraint is provided, so the compiler tries to balance the fMAX-to-II ratio by default.
+In kernel `SchedulerTargetFMAX`, the `[[intel::scheduler_target_fmax_mhz(480)]]` attribute tells the compiler to target 480 MHz. Since the II is unconstrainted, the compiler inserts extra pipelining to schedule the kernel at 480 MHz instead of trying to balance fMAX and II.
+
```

Comment:
Two general questions: 
1. When should I, as a user, use this attribute? 
2. Should we recommend that customers use this flag mainly in combination with the II pragma?

Reply:
> 1. When should I, as a user, use this attribute?

I'll add two sub-sections under `## purpose`: `### Understanding the Tutorial Design` and `### Use cases of intel::scheduler_target_fmax_mhz(N)` to help explain when to use this attribute.

> 2. Should we recommend that customers use this flag mainly in combination with the II pragma?

Yes we should. I was not sure whether should we use other attributes in this tutorial. But once we mention the II attribute tutorial as a prerequisite, I can demonstrate using both of them at the same time.